test/codegen.test.ts

```diff
@@ -0,0 +1,152 @@
+import { join } from 'path';
+import { generate } from '../codegen/src/codegen';
+
+describe('codegen', () => {
+  it('should generate correct TypeScript', async () => {
+    const schemaFilePath = join(__dirname, 'mocks', 'schema.json');
+    const outputFilePath = 'hahaha';
+
+    const writeFile = jest.fn();
+    await generate({ schemaFilePath, outputFilePath, writeFile });
+
+    // Sorry about this.
```

Comment:
What if we use `toMatchSnapshot` instead of adding and using `toMatchWithoutWhitespace`?

We would remove all the expected outputs from this file and it'd be much cleaner.

Reply:
Sure, great idea!